newrelic/newrelic-diagnostics-cli
# Copy over previous work and browser fixtures

# Issue

BrowserAgentGetSource task collects the Browser agent script found in the HTML source page. However, the task should not collect the Browser script if this is found outside of the Head tag element because it hides a potential issue on the customer's end.

# Goals
Improve regex to detect when the Browser script is not in the Head tag.
If it is found somewhere else that it is not the Head, then we should notify the customer about instrumentation issues.

# Implementation Details
The changes have been made in Tasks/Browser/Agent/GetSource.go
We have added some unit tests
Finally, we have enabled the browser suite that previously was commented out in suitesDefinitions.go
 
# How to Test
To test this task, you can build the go binary  with `go build` against this branch, and then run `./newrelic-diagnostics-cli -browser-url https://someurlthathasthenewrelicbrowserscript -suites browser`
In this ticket, you can find a URL that "technically" has injected the browser script, but has done it following bad practices: https://newrelic.zendesk.com/agent/tickets/443022 . So we would expect Diagnostics CLI to give a `failure` status.
This ticket(https://newrelic.zendesk.com/agent/tickets/423509) can provide a URL that has placed the browser script in the right place, so the task should provide a `success` status.
